sequential/seq2pat.py

```diff
@@ -283,12 +294,36 @@ class Seq2Pat:
         sequences but no constraints are used to run the mining efficiently and practically.
         Power users can choose to drop this constraint by setting it to be None or increase the maximum span
         as the system has resources to support.
+    batch_size: Optional[int]
+        When batch_size is set, program runs Seq2Pat on batches of sequences instead of on the entire set of
+        sequences for improving scalability. Each batch contains `batch_size` sequences as a random sample of
+        entire set. A mining task will run on each batch with a reduced minimum row count (min_frequency) threshold.
+        Please refer to description of discount_factor parameter for how min_frequency is reduced.
+        Resulted patterns will be aggregated from the mining results of each batch by calculating the sum of
+        the occurrences. Finally the original minimum row count threshold is applied to the patterns after aggregation.
+    n_jobs: int
+        n_jobs defines the number of processes (n_jobs=2 by default) that are used when mining tasks are applied
+        on batches in parallel. If -1 all CPUs are used. If -2, all CPUs but one are used.
+    seed: int
+        Random seed to make sequences uniformly distributed among batches.
+    discount_factor: float
+        A discount factor is used to reduce the minimum row count (min_frequency) threshold when Seq2Pat is applied
+        on a batch. When min_frequency is a integer, mining task can only be run on the entire set.
+        When min_frequency is a float, mining can be run on batches, with new threshold being defined by
+        max(min_frequency * discount_factor, 1.0/num_rows). num_rows is the number of sequences of one batch.
```

Comment:
isn't num_rows == batch_size? IF so, why not use batch_size instead of introducing num_rows?

Reply:
I am thinking that there would be one batch having number of sequences less than defined batch_size. So I introduce num_rows, which could be more accurate. But this may not be critical or of user's interests. I can use batch_size directly.